L1Trigger/L1THGCal/python/customClustering.py

```diff
@@ -1,238 +1,216 @@
-        [0.020]*BH_DR_GROUP) # BH
-
-
-def custom_2dclustering_distance(process, 
-        distance=6.,# cm
-        seed_threshold=5.,# MipT
-        cluster_threshold=2.# MipT
-        ):
-    parameters_c2d = process.hgcalBackEndLayer1Producer.ProcessorParameters.C2d_parameters
-    parameters_c2d.seeding_threshold_silicon = cms.double(seed_threshold) 
-    parameters_c2d.seeding_threshold_scintillator = cms.double(seed_threshold) 
-    parameters_c2d.clustering_threshold_silicon = cms.double(cluster_threshold) 
-    parameters_c2d.clustering_threshold_scintillator = cms.double(cluster_threshold) 
-    parameters_c2d.dR_cluster = cms.double(distance) 
-    parameters_c2d.clusterType = cms.string('dRC2d') 
+from L1Trigger.L1THGCal.hgcalBackEndLayer1Producer_cfi import dummy_C2d_params, \
+                                                              distance_C2d_params, \
+                                                              topological_C2d_params, \
+                                                              constrTopological_C2d_params
+from L1Trigger.L1THGCal.hgcalBackEndLayer2Producer_cfi import distance_C3d_params, \
+                                                              dbscan_C3d_params, \
+                                                              histoMax_C3d_params, \
+                                                              histoMaxVariableDR_C3d_params, \
+                                                              histoSecondaryMax_C3d_params, \
+                                                              histoInterpolatedMax_C3d_params, \
+                                                              histoThreshold_C3d_params, \
+                                                              dr_layerbylayer, \
+                                                              dr_layerbylayer_Bcoefficient, \
+                                                              neighbour_weights_1stOrder, \
+                                                              neighbour_weights_2ndOrder, \
+                                                              MAX_LAYERS
+
+
+def set_threshold_params(pset, seed_threshold, cluster_threshold):
+    pset.seeding_threshold_silicon = seed_threshold
+    pset.seeding_threshold_scintillator = seed_threshold
+    pset.clustering_threshold_silicon = cluster_threshold
+    pset.clustering_threshold_scintillator = cluster_threshold
+
+
+def custom_2dclustering_distance(process,
+                                 distance=distance_C2d_params.dR_cluster,  # cm
+                                 seed_threshold=distance_C2d_params.seeding_threshold_silicon,  # MipT
+                                 cluster_threshold=distance_C2d_params.clustering_threshold_silicon  # MipT
+                                 ):
+    parameters_c2d = distance_C2d_params.clone()
+    set_threshold_params(parameters_c2d, seed_threshold, cluster_threshold)
+    parameters_c2d.dR_cluster = distance
```

Comment:
parameters can be reset inside the `clone()` statement (everywhere in this file)